sklearn/model_selection/_validation.py

```diff
@@ -989,14 +1053,46 @@ def validation_curve(estimator, X, y, param_name, param_range, groups=None,
     parallel = Parallel(n_jobs=n_jobs, pre_dispatch=pre_dispatch,
                         verbose=verbose)
     out = parallel(delayed(_fit_and_score)(
-        estimator, X, y, scorer, train, test, verbose,
+        estimator, X, y, {'score': scorer}, train, test, verbose,
         parameters={param_name: v}, fit_params=None, return_train_score=True)
         # NOTE do not change order of iteration to allow one time cv splitters
         for train, test in cv.split(X, y, groups) for v in param_range)
+    out = list(zip(*out))
 
-    out = np.asarray(out)
     n_params = len(param_range)
-    n_cv_folds = out.shape[0] // n_params
-    out = out.reshape(n_cv_folds, n_params, 2).transpose((2, 1, 0))
-
+    n_splits = cv.get_n_splits(X, y)
+    out = list(_aggregate_score_dicts(out[i],
+                                      shape=(n_splits, n_params),
+                                      transpose=True)['score'] for i in (0, 1))
     return out[0], out[1]
+
+
+def _aggregate_score_dicts(scores, shape=None, transpose=False):
+    """Aggregate the list of dict to dict of np ndarray
+
+    The aggregated output of _fit_and_score will be a list of dict
+    of form [{'prec': 0.1, 'acc':1.0}, {'prec': 0.1, 'acc':1.0}, ...]
+    Convert it to a dict of array {'prec': np.array([0.1 ...]), ...}
+
+
+    If shape is set, the individual numpy arrays are reshaped accordingly
+
+    If transpose is set, the individual arrays are transposed after reshaping
```

Comment:
add a doctest example here maybe? also, why do we need this again? The ``cv_scores_`` has the dict structure, right? But cross_val_score does not?